### PR TITLE
RzShell: refactor string, regex and byte search

### LINKED ISSUE #4910: rz-bin heap-buffer-overflow
### Work environment

| Questions                                            | Answers
|------------------------------------------------------|--------------------
| OS/arch/bits (mandatory)                             | Ubuntu 20.04.6 LTS
| File format of the file you reverse (mandatory)      | ELF
| Architecture/bits of the file (mandatory)            | x86/64
| `rizin -v` full output, **not truncated** (mandatory)  | rizin 0.8.0 @ linux-x86-64
commit: 309f57434dfa17954f02cdcbb3a2ac4108651767

### Expected behavior
Not segment fault

### Actual behavior
Segment fault (with heap-buffer-overflow)

### Steps to reproduce the behavior
- poc:  [rz-bin-poc-01.zip](https://github.com/user-attachments/files/18817099/rz-bin-poc-01.zip)

- run cmd `rz-bin -z -N":<dH" $poc`

### Additional Logs, screenshots, source code,  configuration dump, ...

./rizin/bins/bin/rz-bin -z -N":<dH" /tmp/poc
=================================================================
==2793982==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x602000183011 at pc 0x7f7c47eaa928 bp 0x7f7c3f8fea40 sp 0x7f7c3f8fea30
WRITE of size 1 at 0x602000183011 thread T1
    #0 0x7f7c47eaa927 in rz_utf8_encode ../librz/util/utf8.c:539
    #1 0x7f7c47e77158 in process_one_string ../librz/util/str_search.c:269
    #2 0x7f7c47e78d5b in rz_scan_strings_raw ../librz/util/str_search.c:523
    #3 0x7f7c43e9510f in string_scan_range ../librz/bin/bfile_string.c:103
    #4 0x7f7c43e95364 in search_string_thread_runner ../librz/bin/bfile_string.c:130
    #5 0x7f7c47e9495a in thread_main_function ../librz/util/thread.c:21
    #6 0x7f7c48411608 in start_thread /build/glibc-FcRMwW/glibc-2.31/nptl/pthread_create.c:477
    #7 0x7f7c4855b352 in __clone (/lib/x86_64-linux-gnu/libc.so.6+0x11f352)

0x602000183011 is located 0 bytes to the right of 1-byte region [0x602000183010,0x602000183011)
allocated by thread T1 here:
    #0 0x7f7c487ffa06 in __interceptor_calloc ../../../../src/libsanitizer/asan/asan_malloc_linux.cc:153
    #1 0x7f7c47e7827d in rz_scan_strings_raw ../librz/util/str_search.c:403
    #2 0x7f7c43e9510f in string_scan_range ../librz/bin/bfile_string.c:103
    #3 0x7f7c43e95364 in search_string_thread_runner ../librz/bin/bfile_string.c:130
    #4 0x7f7c47e9495a in thread_main_function ../librz/util/thread.c:21
    #5 0x7f7c48411608 in start_thread /build/glibc-FcRMwW/glibc-2.31/nptl/pthread_create.c:477

Thread T1 created by T0 here:
    #0 0x7f7c4872c815 in __interceptor_pthread_create ../../../../src/libsanitizer/asan/asan_interceptors.cc:208
    #1 0x7f7c47e94e3a in rz_th_new ../librz/util/thread.c:211
    #2 0x7f7c43e958c9 in create_string_search_thread ../librz/bin/bfile_string.c:202
    #3 0x7f7c43e9792b in rz_bin_file_strings ../librz/bin/bfile_string.c:482
    #4 0x7f7c43eaecc9 in rz_bin_set_and_process_strings ../librz/bin/bobj_process_string.c:26
    #5 0x7f7c43eaacfe in rz_bin_object_process_plugin_data ../librz/bin/bobj_process.c:156
    #6 0x7f7c43ea6f6e in rz_bin_object_new ../librz/bin/bobj.c:529
    #7 0x7f7c43e91667 in rz_bin_file_new_from_buffer ../librz/bin/bfile.c:139
    #8 0x7f7c43e9a4c3 in rz_bin_open_buf ../librz/bin/bin.c:294
    #9 0x7f7c43e9ab52 in rz_bin_open_io ../librz/bin/bin.c:352
    #10 0x7f7c43e99878 in rz_bin_open ../librz/bin/bin.c:233
    #11 0x7f7c48665613 in rz_main_rz_bin ../librz/main/rz-bin.c:1204
    #12 0x55800161b1b4 in main ../binrz/rz-bin/rz-bin.c:8
    #13 0x7f7c48460082 in __libc_start_main ../csu/libc-start.c:308

SUMMARY: AddressSanitizer: heap-buffer-overflow ../librz/util/utf8.c:539 in rz_utf8_encode
Shadow bytes around the buggy address:
  0x0c04800285b0: fa fa fd fa fa fa fd fa fa fa fd fa fa fa fd fa
  0x0c04800285c0: fa fa fd fd fa fa fd fa fa fa fd fa fa fa fd fa
  0x0c04800285d0: fa fa 00 00 fa fa 00 01 fa fa 00 01 fa fa 00 00
  0x0c04800285e0: fa fa fd fd fa fa 00 00 fa fa 00 00 fa fa fa fa
  0x0c04800285f0: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
=>0x0c0480028600: fa fa[01]fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c0480028610: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c0480028620: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c0480028630: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c0480028640: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c0480028650: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
  Shadow gap:              cc
==2793982==ABORTING

### PR DESCRIPTION
**Pull request by Rot127**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Supersedes https://github.com/rizinorg/rizin/pull/4762**

**Detailed description**

## Changes made

- Moves all legacy search commands to RzShell (only commands, inside they still do their string parsing on arguments).
- Refactor string and byte search
 - Move to RzShell
 - Moves: `/` to `/z`.
 - Add support for Unicode and EBCDIC string search.
 - Add support for (Unicode) regex string search.
 - Add support for byte string regex search `/xr`.
 - Add more details to the search help messages.
 - Offsets of the search hits align with the actual encoding. Not with the UTF-8 encoding.
 - Dispatches memory chunks for search into threads.
- Changes to `ps`
  - Adds extra arguments to specify encoding (also EBCDIC).
  - Add additional delimiter argument (stop at first non-printable).
  - Document it more.
  - Add `psu` alias for `ps utf8`
- Changes to Settings
  - Remove `str.search.max_uni_blocks` - Effectively a metric the user should not know about; adds too much complexity.
  - `str.search.max_threads` -> `search.max_threads` - This is a general setting for the search now.
  - `str.search.raw_alignment` -> `search.str.raw_alignment` - Unify settings (only used for RzBin search.).
  - `str.search.encoding` -> `str.encoding` - Valid for all string interpretations.
  - `str.search.min_length` -> `search.str.min_length` - Unify settings.
  - `str.search.buffer_size` -> `search.str.max_length` - Unify settings.
  - `str.search.max_region_size` -> `search.str.max_region_size` - Unify settings.
  - `str.search.check_ascii_freq` -> `search.str.check_ascii_freq` - Unify settings.
- Removed commands
  - `/!` - Because the command modifiers are not properly handled in RzShell yet and the advantage of this one is dubious (IMHO).
  - `/f` - Modifier and obsolete, because search is dispatched into threads.
  - `/b` - Modifier and obsolete, because search is dispatched into threads.
  - `/+` - Because no idea what it does. Seems not particular useful.
  - `/e` - Replaced with regex search in bytes and string search.
  - `/w` - All Unicode is searched now properly with `/z`.
- Make some changes to the string escaping, so it works reliably with Unicode characters.
  - The `RzStrEscOptions` were inconsistently used.
    E.g. show_asciidot (replace non-printable ascii with dot) was ignored for \n, \t etc.
  - Defined Unicode code points are escaped now with `\U00hhhhhh`. All other non-printable bytes are escaped with `\xhh`. There are still some exceptions (when legacy escape functions are used) but most places are ok now.
- General
  - Fix inconsistencies in Unicode decoders/encoders and checkers. They now either return 0 on an invalid decode or the number of bytes the code point requires.
  - Add many unit tests for Unicode related logic.
  - Update Unicode tables to Version 16.
  - Add helper to check code points.
  - Escaped strings now escape valid Unicode code points to `/Uhhhhhh` (if not requested otherwise by the user) and invalid code points to `/xhh`.
  - Add helper functions for hexadecimal strings and bits.

## TODO Overview

**Open issues**

- [x] GCC-12 flag: https://github.com/rizinorg/rizin/commit/6109a0b1eae37e226fd2d5e83829c00f22f7aacb
- [x] Search hits of strings should contain length in characters/code points and graphemes. cc @kazarmy 
- [ ] Add option to define search strategy (graph algos, address window sizes etc.).
- [x] Add option to define what are "unprintable character" exceptions (e.g. `\t`, `\n` etc.).

**Documentation**

- Update book about search:
  - API
  - commands
  - examples
  - edge cases (e.g. [diacritics](https://stackoverflow.com/questions/58770462/how-to-iterate-over-unicode-grapheme-clusters-in-rust), regex search PCRE2)

**Test plan**

Tests were added

**Closing issues**

closes https://github.com/rizinorg/rizin/issues/4910

### CODE REVIEW

**Review by notxvilka — APPROVED** [review]
Looks like YAML generator script needs to be updated to support unicode:
```
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xd0 in position 7330: ordinal not in range(128)
[1947/2315] Compiling C object test/unit/test_agraph.p/test_agraph.c.o
```

**Comment by notxvilka** [comment]
Check also this error for building from a source tarball:
```
Run echo -e "#include <rz_util.h>\nint main(int argc, char **argv) { return rz_str_newf (\"%s\", argv[0]) != NULL? 0: 1; }" > test.c
In file included from test.c:1:
/usr/include/librz/rz_util.h:65:10: fatal error: 'rz_util/rz_unicode.h' file not found
#include "rz_util/rz_unicode.h"
         ^~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

there is a line that needs to be added into the meson file within `librz/include`

**Comment by notxvilka** [comment]
@Rot127 to simplify your debugging, these tests fail also on ARM64 machine: https://app.travis-ci.com/github/rizinorg/rizin/jobs/630766266#L4649

Check also UAF caught by ASAN:
```
[XX] 145 secs db/analysis/golang Resolve all symbols on a linux go1.16 binary (arm32)
RZ_NOPLUGINS=1 /usr/bin/rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -eflirt.sigdb.load.system=false -esearch.show_progress=false -eflirt.sigdb.load.home=false -N -qc 'aalg
fl~sym.go.~?
pdfs @ main ~str.
izq~?
ilq
' bins/golang/btcctl-linux-armv7-v0.22.0-beta
-- stdout
--- expected
+++ actual
@@ -1,159 +1,0 @@
-5124
-0x00270aac str.Failed_to_read_data_from_stdin:__v
-0x00270d70 str.s_command:__v
-0x00270db8 str.1.0
-0x002710ec str.Failed_to_unmarshal_result:__v
-0x0027124c str.Failed_to_format_result:__v
-0x00271338 str.PH2
-0x0027140c str.s_command:__v__code:__s
-0x002714b0 str.The___s__command_can_only_be_used_via_websockets
-0x00271580 str.Unrecognized_command___s
-0x00271608 str.No_command_specified
-0x0027186c str.Chain_Server_Commands:DEBUG_HTTP2
-0x0027187c str.Wallet_Server_Commands____wallet_:
-0x00271ab4 str.PH2
-0x00271bbc str.cannot_use__wallet_with__regtest__btcwallet_not_yet_compatible_with_regtest
-0x00271e10 str.localhost
-0x00272018 str.Use__s__h_to_show_options
-0x002720ec str.btcwallet.conf
-0x00272188 str.Error_creating_a_default_config_file:__v
-0x00272618 str.s:_Multiple_network_params_can_t_be_used_together____choose_one
-0x0027271c str.Error_parsing_config_file:__v
-0x002727f8 str.btcd.conf
-0x00272c0c str.m___s_rpcuser_____s
-0x00272c60 str.m___s_rpcpass_____s
-0x00272cb4 str.m___s_notls__0_1___:_s
-0x00272e4c str.rpcuser__s_rpcpass__s
-0x00272edc str.notls__s
-0x00273460 str.http
-0x0027347c str.:
-0x00273514 str.POST
-0x00273574 str.Content_Type
-0x002735b4 str.application_json
-0x0027373c str.error_reading_json_reply:__v
-0x00273834 str.d__s
-0x002738bc str.s
-0x00273a58 str.https
-0x00273b18 str.d._d._d
-0x00273b50 str.beta
-0x00273c20 str.s__s
-0x00273d74 str.0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz
-0x00273efc str.linux
-0x00273f0c str.btcd
-0x00273f50 str.linux
-0x00273f60 str.btcctl
-0x00273fa4 str.linux
-0x00273fb4 str.btcwallet
-0x0027401c str.btcctl.conf
-0x00274090 str.rpc.cert
-0x00274104 str.rpc.cert
-15370
-bufio
-bytes
-compress/flate
-compress/gzip
-compress/zlib
-container/list
-context
-crypto
-crypto/aes
-crypto/cipher
-crypto/des
-crypto/dsa
-crypto/ecdsa
-crypto/ed25519
-crypto/ed25519/internal/edwards25519
-crypto/elliptic
-crypto/hmac
-crypto/internal/randutil
-crypto/md5
-crypto/rand
-crypto/rc4
-crypto/rsa
-crypto/sha1
-crypto/sha256
-crypto/sha512
-crypto/tls
-crypto/x509
-crypto/x509/pkix
-encoding/asn1
-encoding/base64
-encoding/binary
-encoding/hex
-encoding/json
-encoding/pem
-errors
-fmt
-github.com/btcsuite/btcd/btcec
-github.com/btcsuite/btcd/btcjson
-github.com/btcsuite/btcd/chaincfg
-github.com/btcsuite/btcd/chaincfg/chainhash
-github.com/btcsuite/btcd/txscript
-github.com/btcsuite/btcd/wire
-github.com/btcsuite/btclog
-github.com/btcsuite/btcutil
-github.com/btcsuite/btcutil/base58
-github.com/btcsuite/go-socks/socks
-github.com/jessevdk/go-flags
-golang.org/x/crypto/chacha20
-golang.org/x/crypto/chacha20poly1305
-golang.org/x/crypto/cryptobyte
-golang.org/x/crypto/curve25519
-golang.org/x/crypto/hkdf
-golang.org/x/crypto/poly1305
-golang.org/x/crypto/ripemd160
-golang.org/x/net/dns/dnsmessage
-golang.org/x/net/http/httpguts
-golang.org/x/net/http/httpproxy
-golang.org/x/net/http2/hpack
-golang.org/x/net/idna
-golang.org/x/text/secure/bidirule
-golang.org/x/text/transform
-golang.org/x/text/unicode/bidi
-golang.org/x/text/unicode/norm
-hash/adler32
-hash/crc32
-internal/bytealg
-internal/cpu
-internal/fmtsort
-internal/oserror
-internal/poll
-internal/reflectlite
-internal/singleflight
-internal/syscall/unix
-internal/testlog
-io
-io/fs
-io/ioutil
-log
-math
-math/big
-math/bits
-math/rand
-mime
-mime/multipart
-net
-net/http
-net/http/internal
-net/textproto
-net/url
-os
-os/user
-path
-path/filepath
-reflect
-regexp
-regexp/syntax
-runtime
-runtime/debug
-runtime/internal/atomic
-sort
-strconv
-strings
-sync
-sync/atomic
-syscall
-time
-unicode
-unicode/utf16
-unicode/utf8

-- stderr
=================================================================
==37328==ERROR: AddressSanitizer: heap-use-after-free on address 0x602000142570 at pc 0x7f94fae5ba27 bp 0x7ffe3459c960 sp 0x7ffe3459c108
READ of size 2 at 0x602000142570 thread T0
    #0 0x7f94fae5ba26 in __interceptor_strdup ../../../../src/libsanitizer/asan/asan_interceptors.cpp:452
    #1 0x7f94f7c9ed37 in rz_config_set ../librz/config/config.c:304
    #2 0x7f94eeab4522 in rz_recover_all_golang_functions_strings_handler ../librz/core/cmd/cmd_analysis.c:5906
    #3 0x7f94eeb71b58 in argv_call_cb ../librz/core/cmd/cmd_api.c:742
    #4 0x7f94eeb71b58 in call_cd ../librz/core/cmd/cmd_api.c:801
    #5 0x7f94eeb71b58 in rz_cmd_call_parsed_args ../librz/core/cmd/cmd_api.c:819
    #6 0x7f94eeb3b1e6 in handle_ts_arged_stmt_internal ../librz/core/cmd/cmd.c:3215
    #7 0x7f94eeb3b1e6 in handle_ts_arged_stmt ../librz/core/cmd/cmd.c:3163
    #8 0x7f94eea4bb76 in handle_ts_stmt ../librz/core/cmd/cmd.c:4802
    #9 0x7f94eeaf9865 in handle_ts_statements_internal ../librz/core/cmd/cmd.c:4859
    #10 0x7f94eeaf9865 in handle_ts_statements ../librz/core/cmd/cmd.c:4824
    #11 0x7f94eeafadb2 in core_cmd_tsrzcmd ../librz/core/cmd/cmd.c:4971
    #12 0x7f94eeb11563 in rz_core_cmd_lines ../librz/core/cmd/cmd.c:5027
    #13 0x7f94f92e1e28 in run_commands ../librz/main/rizin.c:355
    #14 0x7f94f92e403b in rz_main_rizin ../librz/main/rizin.c:1435
    #15 0x7f94f8829d8f  (/lib/x86_64-linux-gnu/libc.so.6+0x29d8f)
    #16 0x7f94f8829e3f in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x29e3f)
    #17 0x563676794ab4 in _start (/usr/bin/rizin+0x2ab4)

0x602000142570 is located 0 bytes inside of 4-byte region [0x602000142570,0x602000142574)
freed by thread T0 here:
    #0 0x7f94faeb4537 in __interceptor_free ../../../../src/libsanitizer/asan/asan_malloc_linux.cpp:127
    #1 0x7f94fb7f8053 in rz_syscall_setup ../librz/syscall/syscall.c:250
    #2 0x7f94ee80928c in cb_asmbits ../librz/core/cconfig.c:687
    #3 0x7f94f7ca080d in rz_config_set_i ../librz/config/config.c:457
    #4 0x7f94ee85f6fc in rz_core_seek_arch_bits ../librz/core/cio.c:137
    #5 0x7f94f431ad88 in rz_analysis_op ../librz/arch/op.c:119
    #6 0x7f94ee9341b3 in decode_val_set_size ../librz/core/golang.c:841
    #7 0x7f94ee937a64 in go_is_sign_match ../librz/core/golang.c:794
    #8 0x7f94ee93b95d in golang_recover_string_arm32 ../librz/core/golang.c:1261
    #9 0x7f94ee945022 in rz_core_analysis_resolve_golang_strings ../librz/core/golang.c:1952
    #10 0x7f94eeab44f9 in rz_recover_all_golang_functions_strings_handler ../librz/core/cmd/cmd_analysis.c:5905
    #11 0x7f94eeb71b58 in argv_call_cb ../librz/core/cmd/cmd_api.c:742
    #12 0x7f94eeb71b58 in call_cd ../librz/core/cmd/cmd_api.c:801
    #13 0x7f94eeb71b58 in rz_cmd_call_parsed_args ../librz/core/cmd/cmd_api.c:819
    #14 0x7f94eeb3b1e6 in handle_ts_arged_stmt_internal ../librz/core/cmd/cmd.c:3215
    #15 0x7f94eeb3b1e6 in handle_ts_arged_stmt ../librz/core/cmd/cmd.c:3163
    #16 0x7f94eea4bb76 in handle_ts_stmt ../librz/core/cmd/cmd.c:4802
    #17 0x7f94eeaf9865 in handle_ts_statements_internal ../librz/core/cmd/cmd.c:4859
    #18 0x7f94eeaf9865 in handle_ts_statements ../librz/core/cmd/cmd.c:4824
    #19 0x7f94eeafadb2 in core_cmd_tsrzcmd ../librz/core/cmd/cmd.c:4971
    #20 0x7f94eeb11563 in rz_core_cmd_lines ../librz/core/cmd/cmd.c:5027
    #21 0x7f94f92e1e28 in run_commands ../librz/main/rizin.c:355
    #22 0x7f94f92e403b in rz_main_rizin ../librz/main/rizin.c:1435
    #23 0x7f94f8829d8f  (/lib/x86_64-linux-gnu/libc.so.6+0x29d8f)

previously allocated by thread T0 here:
    #0 0x7f94fae5b9a7 in __interceptor_strdup ../../../../src/libsanitizer/asan/asan_interceptors.cpp:454
    #1 0x7f94fb7f805b in rz_syscall_setup ../librz/syscall/syscall.c:251
    #2 0x7f94ee80928c in cb_asmbits ../librz/core/cconfig.c:687
    #3 0x7f94f7ca080d in rz_config_set_i ../librz/config/config.c:457
    #4 0x7f94ee85f6fc in rz_core_seek_arch_bits ../librz/core/cio.c:137
    #5 0x7f94f431ad88 in rz_analysis_op ../librz/arch/op.c:119
    #6 0x7f94ee9341b3 in decode_val_set_size ../librz/core/golang.c:841
    #7 0x7f94ee937a64 in go_is_sign_match ../librz/core/golang.c:794
    #8 0x7f94ee93b95d in golang_recover_string_arm32 ../librz/core/golang.c:1261
    #9 0x7f94ee945022 in rz_core_analysis_resolve_golang_strings ../librz/core/golang.c:1952
    #10 0x7f94eeab44f9 in rz_recover_all_golang_functions_strings_handler ../librz/core/cmd/cmd_analysis.c:5905
    #11 0x7f94eeb71b58 in argv_call_cb ../librz/core/cmd/cmd_api.c:742
    #12 0x7f94eeb71b58 in call_cd ../librz/core/cmd/cmd_api.c:801
    #13 0x7f94eeb71b58 in rz_cmd_call_parsed_args ../librz/core/cmd/cmd_api.c:819
    #14 0x7f94eeb3b1e6 in handle_ts_arged_stmt_internal ../librz/core/cmd/cmd.c:3215
    #15 0x7f94eeb3b1e6 in handle_ts_arged_stmt ../librz/core/cmd/cmd.c:3163
    #16 0x7f94eea4bb76 in handle_ts_stmt ../librz/core/cmd/cmd.c:4802
    #17 0x7f94eeaf9865 in handle_ts_statements_internal ../librz/core/cmd/cmd.c:4859
    #18 0x7f94eeaf9865 in handle_ts_statements ../librz/core/cmd/cmd.c:4824
    #19 0x7f94eeafadb2 in core_cmd_tsrzcmd ../librz/core/cmd/cmd.c:4971
    #20 0x7f94eeb11563 in rz_core_cmd_lines ../librz/core/cmd/cmd.c:5027
    #21 0x7f94f92e1e28 in run_commands ../librz/main/rizin.c:355
    #22 0x7f94f92e403b in rz_main_rizin ../librz/main/rizin.c:1435
    #23 0x7f94f8829d8f  (/lib/x86_64-linux-gnu/libc.so.6+0x29d8f)

SUMMARY: AddressSanitizer: heap-use-after-free ../../../../src/libsanitizer/asan/asan_interceptors.cpp:452 in __interceptor_strdup
Shadow bytes around the buggy address:
  0x0c0480020450: fa fa 00 07 fa fa 06 fa fa fa 00 07 fa fa 00 05
  0x0c0480020460: fa fa fa fa fa fa 00 05 fa fa 05 fa fa fa 00 04
  0x0c0480020470: fa fa 00 01 fa fa 00 04 fa fa 00 01 fa fa 00 00
  0x0c0480020480: fa fa 00 fa fa fa 00 00 fa fa 06 fa fa fa 00 00
  0x0c0480020490: fa fa 00 00 fa fa 02 fa fa fa 00 00 fa fa 00 06
=>0x0c04800204a0: fa fa 06 fa fa fa 00 06 fa fa 00 05 fa fa[fd]fa
  0x0c04800204b0: fa fa 00 05 fa fa 05 fa fa fa 00 00 fa fa 05 fa
  0x0c04800204c0: fa fa 00 00 fa fa 05 fa fa fa 00 05 fa fa 06 fa
  0x0c04800204d0: fa fa 00 05 fa fa 00 05 fa fa 02 fa fa fa 00 05
  0x0c04800204e0: fa fa fa fa fa fa 00 00 fa fa fa fa fa fa 00 00
  0x0c04800204f0: fa fa 00 05 fa fa fa fa fa fa 00 05 fa fa 00 fa
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07 
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
  Shadow gap:              cc
==37328==ABORTING

-- exit status: 1
```
https://github.com/rizinorg/rizin/actions/runs/13443445156/job/37563159709?pr=4919#step:11:2556

**Comment by Rot127** [comment]
Black error: https://github.com/rizinorg/rizin/pull/4921

**Review by notxvilka — APPROVED** [review]
LGTM.
